src/EditorFeatures/CSharpTest/AddBraces/AddBracesTests.cs

```diff
@@ -342,13 +443,192 @@ static void Main()
             return;
         }
     }
-}");
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, false)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        public async Task FireForIfNestedInElseWithoutBracesWithMultilineContext1(int bracesPreference, bool expectDiagnostic)
+        {
+            await TestMissingOrInRegularAndScriptAsync(
+            @"
+class Program
+{
+    static void Main()
+    {
+        if (true)
+            if (true)   // This multiline statement does not directly impact the other nested statement
+                return;
+            else
+                return;
+        else [|if|] (false) return;
+    }
+}",
+
+   @"
+class Program
+{
+    static void Main()
+    {
+        if (true)
+            if (true)   // This multiline statement does not directly impact the other nested statement
+                return;
+            else
+                return;
+        else if (false)
+        {
+            return;
+        }
+    }
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
+        }
+
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, true)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        public async Task FireForIfNestedInElseWithoutBracesWithMultilineContext2(int bracesPreference, bool expectDiagnostic)
+        {
+            await TestMissingOrInRegularAndScriptAsync(
```

Comment:
i'm finding these tests fairly confusing sinc ethis is called TestMissingOrInRegularAndScriptAsync, but then shows an expected result.

Reply:
Agreed this could be confusing to read. I'll find a way to rename it so it doesn't start with `TestMissing`.

Reply:
➡️ Renamed to `TestAsync`